Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/BoundingBoxes/BoundingBox.cs

```diff
@@ -0,0 +1,640 @@
+                }
+
+                rendererObj.bounds.GetCornerPositionsFromRendererBounds(ref corners);
+                boundsPoints.AddRange(corners);
+            }
+        }
+
+        public static Bounds GetRenderBounds(GameObject target)
+        {
+            Bounds bounds = new Bounds();
+            Renderer[] renderers = target.GetComponentsInChildren<Renderer>();
+            for (int i = 0; i < renderers.Length; ++i)
+            {
+                var rendererObj = renderers[i];
+                bounds.ExpandToContain(rendererObj.bounds);
+            }
+
+            return bounds;
+        }
+
+        /// <summary>
+        /// GetMeshFilterBoundsPoints - gets boundingbox points using MeshFilter method.
+        /// </summary>
+        /// <param name="target">gameObject that boundingbox bounds</param>
+        /// <param name="boundsPoints">array reference that gets filled with points</param>
+        /// <param name="ignoreLayers">layerMask to simplify search</param>
+        public static void GetMeshFilterBoundsPoints(GameObject target, List<Vector3> boundsPoints, LayerMask ignoreLayers)
+        {
+            MeshFilter[] meshFilters = target.GetComponentsInChildren<MeshFilter>();
+            for (int i = 0; i < meshFilters.Length; i++)
+            {
+                var meshFilterObj = meshFilters[i];
+                if (ignoreLayers == (1 << meshFilterObj.gameObject.layer | ignoreLayers))
+                {
+                    continue;
+                }
+
+                Bounds meshBounds = meshFilterObj.sharedMesh.bounds;
+                meshBounds.GetCornerPositions(meshFilterObj.transform, ref corners);
+                boundsPoints.AddRange(corners);
+            }
+            RectTransform[] rectTransforms = target.GetComponentsInChildren<RectTransform>();
+            for (int i = 0; i < rectTransforms.Length; i++)
+            {
+                rectTransforms[i].GetWorldCorners(rectTransformCorners);
+                boundsPoints.AddRange(rectTransformCorners);
+            }
+        }
+
+        public static void GetNonAxisAlignedBB_Corners(GameObject target, List<Vector3> boundsPoints)
+        {
+            LayerMask mask = new LayerMask();
+
+            GameObject clone = GameObject.Instantiate(target);
+            clone.transform.localRotation = Quaternion.identity;
+            clone.transform.position = Vector3.zero;
+            clone.transform.localScale = Vector3.one;
+            BoundingBox.GetMeshFilterBoundsPoints(clone, boundsPoints, mask);
+            Vector3 centroid = target.transform.position;
+            GameObject.Destroy(clone);
+
+            for (int i = 0; i < boundsPoints.Count; ++i)
+            {
+                boundsPoints[i] = target.transform.localToWorldMatrix.MultiplyPoint(boundsPoints[i]);
+            }
+        }
+
+        private static Vector3[] corners = null;
+        private static Vector3[] rectTransformCorners = new Vector3[4];
+        #endregion
```

Comment:
I used these static methods all the time in HTK. You should put them into a static extensions class. So helpful for creating bounds based on an object (and its children).